x/auth/ante/setup.go

```diff
@@ -39,11 +45,17 @@ func (sud SetUpContextDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, _ bool,
 
 	newCtx = SetGasMeter(ctx, gasTx.GetGas())
 
-	if cp := ctx.ConsensusParams(); cp.Block != nil {
+	// TODO: possibly cache the result of this query for other antehandlers to use
```

Comment:
In another pr we are working on caching the decoded object to avoid extra over head of decoding each time. If the value is on the context we get it from there. So the overhead should be minimal.